htdocs/js/ui/notebook_protection_controller.js

```diff
@@ -0,0 +1,620 @@
+			$scope.allUsersGroups = [];
+
+			//value of the select dropdown
+			$scope.selectedGroup1 = '';
+
+			//tab/state
+			$scope.currentTab = 1;
+
+			//value of the select dropdown
+			$scope.selectedGroup2 = '';
+
+			//group admins
+			$scope.groupAdmins = [];
+			//group members
+			$scope.groupMembers = [];
+
+			//checks and balances
+			$scope.originalAdmins = [];
+			$scope.originalMembers = [];
+
+			//way to stop watching collections
+			$scope.theWatcherAdmins = null;
+			$scope.theWatcherMembers = null;
+
+		}
+
+	
+
+		$scope.saveNotebookTab = function(){
+
+			//console.log($scope.filterCondition.operator);
+
+			//if making private of moving to another group
+			if($scope.belongsToGroup){
+
+				var conf = confirm("Are you sure you want to move your notebook to this group?");
+				if(conf){
+					console.log('notebook id is '+$scope.notebookGistName);
+					GroupsService.setNotebookGroup($scope.notebookGistName, $scope.selectedGroup1.id )
+					.then(function(data){
+						console.log('data is '+data);
+						//need to update the notebook's tip to reflect the change
+						$scope.cancel();
+
+					})
+					.catch(function(e){
+						console.log(e)
+					})
+				}
+			}
+			else{
+
+				var conf = confirm("Are you sure you want to make your notebook public?");
+				if(conf){
+					console.log('notebook id is '+$scope.notebookGistName);
+					GroupsService.setNotebookGroup($scope.notebookGistName, null )
+					.then(function(data){
+						console.log('data is '+data);
+					})
+					.catch(function(e){
+						
+						$scope.cancel();
+
+						console.log(e)
+					})
+				}
+			}
+		}
+
+		
+
+		//GROUP TAB METHODS
+		/////////////////////////////////////////
+
+	    $scope.createGroup = function(){
+	    	var pr = prompt("Enter new group name", "");
+
+	    	if (pr != null) {
+
+                GroupsService.createGroup(pr)
+                .then(function(data){
+
+                    $scope.$evalAsync(function(){
+
+                    	$scope.allUsersGroups.push({
+                    		id:data,
+                    		name:pr,
+                    		isAdmin:true
+                    	});
+                    });
+                    console.log('group created '+data);
+                })
+                .catch(function(e){
+                    alert(e);
+                })
+	            
+			}
+	    }
+
+	    $scope.renameGroup = function(){
+	    	var pr = prompt("Enter new group name", $scope.selectedGroup2.name);
+
+	    	if (pr != null) {
+				var r = confirm('Are you sure you want to rename "'+$scope.selectedGroup2.name+' to "'+pr+'"?');
+
+				if(r === true){
+
+					//var groupId;
+	                var prevGroupId = $scope.selectedGroup2.id;
+	                GroupsService.changeGroupName($scope.selectedGroup2.id ,pr)
+	                .then(function(data){
+	                    //groupId = data;
+
+	                    var theIndex;
+						for(var i = 0; i < $scope.allUsersGroups.length; i ++){
```

Comment:
using something like `_.find` would reduce the amount of code. I'm seeing this loop repeated quite a few times.